src/lib/encryption.ts

```diff
@@ -29,7 +29,9 @@ type CipherText = {
 
 function encrypt(message: Field[], otherPublicKey: PublicKey) {
   // key exchange
-  let privateKey = Scalar.random();
+  let privateKey = Circuit.inCheckedComputation()
+    ? Circuit.witness(Scalar, () => Scalar.random())
+    : Scalar.random();
```

Comment:
Nice! IMO `Scalar.random` should do that out of the box, because someone else might make the same mistake